src/utilities/sql/SqlFile.hpp

```diff
@@ -816,26 +817,79 @@ class UTILITIES_API SqlFile {
   /** @name Generic Query Interface */
   //@{
 
+
   /// execute a statement and return the first (if any) value as a double
-  boost::optional<double> execAndReturnFirstDouble(const std::string& statement) const;
+  // Variadic arguments are the bind arguments if any, to replace '?' placeholders in the statement string
+  template<typename... Args>
+  boost::optional<double> execAndReturnFirstDouble(const std::string& statement, Args&& ... args) const {
+    boost::optional<double> result;
+    if (m_impl){
+      result = m_impl->execAndReturnFirstDouble(statement, std::forward<Args>(args)...);
```

Comment:
Not sure whether I should use `std::forward<Args>(args)...` here actually. Same comment as in SqlFile_Impl.hpp